### PR TITLE
[search] Refactored the lang indexes in the KeywordLangMatcher.

### PR DESCRIPTION
**Pull request by mpimenov**
The efficiency should drop a little because now new vectors
are created all the time but, taking into account how small
a part of processing a query this is, I think that introducing
additional lighter methods is not justified.

### CODE REVIEW

File: search/keyword_lang_matcher.hpp [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
+#include <utility>
+#include <vector>
 
 namespace search
 {
-
 class KeywordLangMatcher
 {
 public:
+  using StringT = KeywordMatcher::StringT;
```
Comment (ygorshenin) [inline search/keyword_lang_matcher.hpp]: Is it possible to get rid of "T" suffixes?

Comment (mpimenov) [reply search/keyword_lang_matcher.hpp]: done

File: search/keyword_matcher.cpp [hunk 2]
```diff
@@ -99,13 +98,17 @@ KeywordMatcher::ScoreT KeywordMatcher::Score(StringT const * tokens, size_t coun
...
 
-KeywordMatcher::ScoreT::ScoreT()
-  : m_sumTokenMatchDistance(0), m_nameTokensMatched(0), m_nameTokensLength(0),
-    m_numQueryTokensAndPrefixMatched(0), m_bFullQueryMatched(false), m_bPrefixMatched(false)
+KeywordMatcher::Score::Score()
+  : m_sumTokenMatchDistance(0)
+  , m_nameTokensMatched(0)
+  , m_nameTokensLength(0)
+  , m_numQueryTokensAndPrefixMatched(0)
+  , m_bFullQueryMatched(false)
```
Comment (ygorshenin) [inline search/keyword_matcher.cpp]: How about to get rid of this "b" prefixes?

Comment (mpimenov) [reply search/keyword_matcher.cpp]: done

**Pull request merged**